File: 2024-1/ejercicios-java/clase-02/Persona.java
class Persona {
	private int dni;
	private char sexo;
	private String nombre;
	private double sueldo;

	public Persona(
		int dni, char sexo, String nombre, double sueldo) {

		this.dni = dni;
		this.sexo = sexo;
		this.nombre = nombre;
		this.sueldo = sueldo;
	}

	public void imprimirDatos() {
		System.out.println(
				String.format("%d, %c, %s, %.2f", 
								dni, sexo, nombre, sueldo));
	}

	public int getDni() {
		return dni;
	}

	public char getSexo() {
		return sexo;
	}

	public String getNombre() {
		return nombre;
	}

	public double getSueldo() {
		return sueldo;
	}

	public void setDni(int dni) {
		this.dni = dni;
	}

	public void setSexo(char sexo) {
		this.sexo = sexo;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public void setSueldo(double sueldo) {
		this.sueldo = sueldo;
	}
}

File: 2026-1/capitulo-03/herencia/ejercicio02/Persona.cpp
//
// Created by Eric Huiza on 5/15/26.
//

#include "Persona.h"

Persona::Persona() {
    dni = 0;
    nombre = nullptr;
}

Persona::Persona(const Persona& origin) : Persona() {
    *this = origin;
}

int Persona::getDni() const {
    return dni;
}

char* Persona::getNombre() const {
    return nombre;
}

void Persona::setDni(int dni) {
    this->dni = dni;
}

void Persona::setNombre(char* nombre) {
    setCadena(this->nombre, nombre);
}

Persona& Persona::operator=(const Persona& origin) {
    if (this != &origin) {
        this->setDni(origin.getDni());
        this->setNombre(origin.getNombre());
    }
    return *this;
}

ifstream& Persona::leer(ifstream& archivo) {
    int dni;
    if (archivo >> dni) {
        archivo.ignore();
        char* nombre = leerCadena(archivo);

        this->setDni(dni);
        this->setNombre(nombre);

        delete[] nombre;
    }

    return archivo;
}

ofstream& Persona::imprimir(ofstream& archivo) const {
    archivo << "DNI: " << this->getDni() << endl;
    archivo << "Nombre: " << this->getNombre() << endl;

    return archivo;
}

Persona::~Persona() {
    delete[] this->nombre;
}

ifstream& operator>>(ifstream& archivo, Persona& persona) {
    return persona.leer(archivo);
}

ofstream& operator<<(ofstream& archivo, const Persona& persona) {
    return persona.imprimir(archivo);
}

